### PR TITLE
Initial implementation of gsc-ec-converter.

### PR DESCRIPTION
**Pull request by itohro**
Tested the converter against 2 projects that have gs-collections as dependencies.

https://github.com/jClarity/slow-crud-app
https://github.com/sMiyoshi/java-collections-sample

### CODE REVIEW

File: src/main/java/org/eclipse/collections/tools/Converter.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    {
+        try
+        {
+            Files.walk(Paths.get(path))
+                    .filter(Files::isRegularFile)
+                    .filter(file -> IntInterval.fromTo(0, file.getNameCount() - 1).collect(file::getName).noneSatisfy(s -> s.toString().startsWith(".")))
+                    .forEach(file -> {
+                        try
+                        {
+                            MutableList<String> allLines = ListAdapter.adapt(Files.readAllLines(file, Charset.defaultCharset()));
```
Comment (gs-rezaem) [inline src/main/java/org/eclipse/collections/tools/Converter.java]: This is not production quality code, but this is not production code, so it's fine.
Typically, reading an entire file into memory to replace strings in it is not the right approach. Line by line would be the right way.

File: src/main/java/org/eclipse/collections/tools/Converter.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+package org.eclipse.collections.tools;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.MalformedInputException;
+import java.nio.file.Files;
```
Comment (gs-rezaem) [inline src/main/java/org/eclipse/collections/tools/Converter.java]: This requires JDK 7 to compile/run. Is that ok?

File: src/main/java/org/eclipse/collections/tools/Converter.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        }
+    }
+
+    private static void convert(String path)
+    {
+        try
+        {
+            Files.walk(Paths.get(path))
+                    .filter(Files::isRegularFile)
+                    .filter(file -> IntInterval.fromTo(0, file.getNameCount() - 1).collect(file::getName).noneSatisfy(s -> s.toString().startsWith(".")))
```
Comment (gs-rezaem) [inline src/main/java/org/eclipse/collections/tools/Converter.java]: why are we filtering out "." files instead of just looking for ".java" files?

**Comment by itohro** [comment]
Thanks for the feedback, I've updated below.

* Improved the file read part to stream line by line and update.
* Added files suffix scope .java, .xml and .gradle. Also updated README.md accordingly.

It should be fine to limit to JDK 7 and upwards. This application is meant to run standalone and it's reasonable to discard JDK 6 and below at this point. We can consider if any request is raised to support JDK 6 and below.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,6 @@
+build
+.gradle
+gradle.prefs
+.idea/workspace.xml
+.idea/uiDesigner.xml
+src/test/resources/test
```
Comment (motlin) [inline .gitignore]: Why exclude a directory under src/test/resources?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
+Eclipse Collections converter 
+=============================
+
+A converter application to replace [GS Collections](https://github.com/goldmansachs/gs-collections) dependencies with [Eclipse Collections](https://www.eclipse.org/collections/).
+
+GS Collections 7.0 and Eclipse Collections 7.0 are functionally equivalent besides the name space changes below.
+
+* Maven groupId: from ```com.goldmansacks``` to ```org.eclipse.collections```
```
Comment (motlin) [inline README.md]: typo in the groupId

File: README.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+* Maven artifactId: from ```gs-collections``` to ```eclipse-collections```
+* Java package: from ```com.gs.collections``` to ```org.eclipse.collections```
+
+To help users migrating from GS Collections to Eclipse Collections, 
+we created a handy converter to replace the dependencies in your project.
+
```
Comment (motlin) [inline README.md]: Not just dependencies - alters source code in place.

File: build.gradle [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompany this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+group 'org.eclipse.collections'
+version '7.1.0-SNAPSHOT'
```
Comment (motlin) [inline build.gradle]: Should this use a production release (7.0.0) instead of 7.1.0-SNAPSHOT?

**Comment by itohro** [comment]
As part of unit test, test files are copied to ```src/test/resources/test```, main method runs against this dir, and cleaned-up afterwards. Added this folder to .gitignore just to make sure anything under this folder is accidentally committed.

Thanks for the feedback in README.md. I've committed changes with fixup so they're incorporated in 657afde.

**Comment by motlin** [comment]
Is it possible to use the JUnit Rule for temporary folders instead? Or
target/

**Comment by itohro** [comment]
Good suggestion indeed. I've refactored the test to use temp dir.

**Pull request merged**